### PR TITLE
Added metadata to tap.py for Tap Plots

### PR DESCRIPTION
**Pull request by BhaswatiRoy**
Fix: https://github.com/bokeh/bokeh/issues/11765

Added metadata for Tap Plots in tap.py file

### CODE REVIEW

**Comment by bryevdv** [comment]
Thanks for the PR @BhaswatiRoy  FYI @tcmetzger has indicated he should be able to review this PR by the end of the week

**Comment by BhaswatiRoy** [comment]
> Thanks for the PR @BhaswatiRoy FYI @tcmetzger has indicated he should be able to review this PR by the end of the week

thanks @bryevdv for informing

File: examples/plotting/tap.py [hunk 1]
```diff
@@ -1,3 +1,14 @@
+''' A interactive numerical tap plot based on simple Python array of data.
```
Comment (tcmetzger) [inline examples/plotting/tap.py]:
```suggestion
''' An interactive numerical tap plot based on a simple Python array of data.
```

File: examples/plotting/tap.py [hunk 2]
```diff
@@ -1,3 +1,14 @@
...
+    One particiular datapoint can be focused by tapping on it.
```
Comment (tcmetzger) [inline examples/plotting/tap.py]:
```suggestion
    You can select any datapoint by tapping on it. 
```

File: examples/plotting/tap.py [hunk 3]
```diff
@@ -1,3 +1,14 @@
+''' A interactive numerical tap plot based on simple Python array of data.
+    One particiular datapoint can be focused by tapping on it. 
+    This highlights that datapoint, and displays other datapoints in faded color.
```
Comment (tcmetzger) [inline examples/plotting/tap.py]:
```suggestion
    This highlights that datapoint and displays all other datapoints in a faded color.
```

**Review by tcmetzger — CHANGES_REQUESTED** [review]
Thank you, I left some small comments in the review!

**Comment by BhaswatiRoy** [comment]
Thanks @tcmetzger for the suggestions, I have done the required changes.

File: examples/plotting/tap.py [hunk 4]
```diff
@@ -1,3 +1,14 @@
+''' An interactive numerical tap plot based on a simple Python array of data.
+    You can select any datapoint by tapping on it. 
+    This highlights that datapoint and displays all other datapoints in a faded color.
+
+.. bokeh-example-metadata::
+    :apis: bokeh.plotting.figure, bokeh.plotting.show, bokeh.models.TapTool
+    :refs: :ref:`ug_basic_tap`
```
Comment (tcmetzger) [inline examples/plotting/tap.py]:
```suggestion
    :refs: :ref:`ug_interaction_tools_clicktap`
```
I just now checked the ref - this needs to be an actual reference from the RST documentation. This is a bit obscure sometimes, unfortunately! In this case, the reference is from `tools.rst`, linking to https://docs.bokeh.org/en/latest/docs/user_guide/interaction/tools.html#click-tap-tools

**Review by tcmetzger — CHANGES_REQUESTED** [review]
Sorry, there is one more thing that I overlooked earlier!

**Comment by BhaswatiRoy** [comment]
Thanks for pointing that out @tcmetzger, I have corrected that.

File: examples/plotting/tap.py [hunk 5]
```diff
@@ -1,3 +1,14 @@
+''' An interactive numerical tap plot based on a simple Python array of data.
+    You can select any datapoint by tapping on it. 
+    This highlights that datapoint and displays all other datapoints in a faded color.
+
+.. bokeh-example-metadata::
+    :apis: bokeh.plotting.figure, bokeh.plotting.show, bokeh.models.TapTool
```
Comment (tcmetzger) [inline examples/plotting/tap.py]:
```suggestion
    :apis: bokeh.plotting.figure, bokeh.io.show, bokeh.models.TapTool
```
And this one needs to be for https://docs.bokeh.org/en/latest/docs/reference/io.html#bokeh.io.show - sorry for the multiple requests!

**Review by tcmetzger — CHANGES_REQUESTED** [review]

File: examples/plotting/tap.py (hunk 5 above)
Comment (tcmetzger) [reply examples/plotting/tap.py]: The best way to verify these api paths is looking up the method or class in the reference guide https://docs.bokeh.org/en/latest/docs/reference.html

**Comment by BhaswatiRoy** [comment]
done!!

**Comment by tcmetzger** [comment]
Thank you!!

**Review by tcmetzger — APPROVED** [review]

**Pull request merged**